design-documents/graph-ql/coverage/b2b/negotiableQuotes.graphqls

```diff
@@ -157,7 +157,29 @@ input DeleteNegotiableQuotesInput {
     quote_uids: [ID!]! @doc(description: "A List of UIDs obtained from negotiable quote types")
 }
 
+type NegotiableQuoteOperationSuccess {
+    quote_uid: ID!
+}
+
+interface NegotiableQuoteOperationErrorInterface {
+    quote_uid: ID!
+    message: String
+}
+
+type NegotiableQuoteInvalidStateError implements NegotiableQuoteOperationErrorInterface {
+}
+
+type NegotiableQuoteNotFoundError implements NegotiableQuoteOperationErrorInterface {
+}
+
+type NegotiableQuoteUndefinedError implements NegotiableQuoteOperationErrorInterface {
+}
+
+union NegotiableQuoteOperationResult = NegotiableQuoteOperationSuccess | NegotiableQuoteOperationErrorInterface
+
 type DeleteNegotiableQuotesOutput {
+    result_status: TriStateMutationResultStatus!
+    operation_results: [NegotiableQuoteOperationResult!]!
```

Comment:
Just to be clear, if the result_status is a SUCCESS, then we don't really need to concern ourselves with the individual operation_results. Is that right?

Reply:
Correct

Reply:
Thoughts on renaming this field to `quote_operation_results`?

Reply:
Hmm, I think I prefer leaving it as `operation_results`.

Reply:
Looks like you're proposing a single array containing a mixture of successes and errors. We had discussed having two lists, separated into successes and errors, but one list will work as well. It would be nice if `NegotiableQuoteOperationResult` included a status as well. As proposed, we'd have to use `message` as a heuristic, right?

```jsx
const elements = []

for (const result of data.operation_results) {
  const { quote_uid: id, message } = result
  const hasError = message != null // heuristic

  // ...
}
```

Reply:
The idea behind the Union type is that you can filter on the type of the result; it will either be an instance of `NegotiableQuoteOperationSuccess` or `NegotiableQuoteOperationFailureInterface`.  See the article I posted previously talking about this strategy: https://github.com/magento/architecture/pull/505#issuecomment-964570720

Reply:
Yeah we talked about having the separate lists, but after you dropped off we continued and decided the unified list made more sense and should be just as easy to work with.